hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupManifest.java

```diff
@@ -512,11 +512,11 @@ public void addDependentImage(BackupImage image) {
    * Set the incremental timestamp map directly.
    * @param incrTimestampMap timestamp map
    */
-  public void setIncrTimestampMap(HashMap<TableName, HashMap<String, Long>> incrTimestampMap) {
+  public void setIncrTimestampMap(Map<TableName, Map<String, Long>> incrTimestampMap) {
     this.backupImage.setIncrTimeRanges(incrTimestampMap);
   }
 
-  public Map<TableName, HashMap<String, Long>> getIncrTimestampMap() {
+  public Map<TableName, Map<String, Long>> getIncrTimestampMap() {
```

Comment:
Might have been better to do these converstions -- HashMap to Map -- as an issue of their own because they distract from meat of this PR (They are good, just off-the-point I'd say). Next time.

Reply:
Sure. Will keep this on mind.

For this particular thing, it was related. For BackupLogCleaner `getIncrTimestampMap` was to be stored as part of BackupInfo and restored. Proto deser was resulting into Type mismatch between Map and HashMap. So, I had to club this as part of same PR.